applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1542,43 +1543,35 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateExtrapolationMatrix(Bounde
 template <>
 void UPwSmallStrainElement<2, 3>::CalculateExtrapolationMatrix(BoundedMatrix<double, 3, 3>& rExtrapolationMatrix)
 {
```

Comment:
I was wondering is it better to move the content to the template instead of these two instantiations?  Right now there is only difference between them in TNumNodes value. The template can check TNumNodes values to avoid using it for other values. What is your opinion?

Reply:
I think template is redundant here as no argument is passed to template. Can the template be removed from here?

Reply:
Nice suggestion. I have moved the code of the template specializations to the template itself, using `if constexpr` to distinguish the supported cases. Since this is a pure refactoring, the function's behavior should not change. Therefore, the tested conditions reflect the earlier template specializations. To make the code easier to read, I have also extracted two local helper functions that return extrapolation matrices for linear tetrahedra and linear hexahedra.